app/portainer/views/home/homeController.js

```diff
@@ -1,95 +1,132 @@
 angular.module('portainer.app')
-.controller('HomeController', ['$q', '$scope', '$state', 'Authentication', 'EndpointService', 'EndpointHelper', 'GroupService', 'Notifications', 'EndpointProvider', 'StateManager', 'ExtensionManager', 'ModalService', 'MotdService',
-function ($q, $scope, $state, Authentication, EndpointService, EndpointHelper, GroupService, Notifications, EndpointProvider, StateManager, ExtensionManager, ModalService, MotdService) {
+  .controller('HomeController', ['$q', '$scope', '$state', 'Authentication', 'EndpointService', 'EndpointHelper', 'GroupService', 'Notifications', 'EndpointProvider', 'StateManager', 'ExtensionManager', 'ModalService', 'MotdService',
+    function ($q, $scope, $state, Authentication, EndpointService, EndpointHelper, GroupService, Notifications, EndpointProvider, StateManager, ExtensionManager, ModalService, MotdService) {
 
-  $scope.goToDashboard = function(endpoint) {
-    EndpointProvider.setEndpointID(endpoint.Id);
-    EndpointProvider.setEndpointPublicURL(endpoint.PublicURL);
-    if (endpoint.Type === 3) {
-      switchToAzureEndpoint(endpoint);
-    } else {
-      switchToDockerEndpoint(endpoint);
-    }
-  };
+      $scope.goToDashboard = function (endpoint) {
+        var status = 1;
+
+        EndpointService.checkEndpointStatus(endpoint.Id)
```

Comment:
See my comment above about using SystemService.ping here instead of this function that is simply wrapping it.